File: server/src/main/java/com/ssafy/cafe/model/service/CommentService.java
package com.ssafy.cafe.model.service;

import java.util.List;
import com.ssafy.cafe.model.dto.Comment;

public interface CommentService {
    /**
     * Comment를 등록한다.
     * @param comment
     */
    void addComment(Comment comment);
    
    /**
     * id에 해당하는 comment를 반환한다.
     * @param id
     * @return
     */
    Comment selectComment(Integer id);
    
    /**
     * id에 해당하는 Comment를 삭제한다.
     * @param id
     */
    void removeComment(Integer id);
    
   /**
    * Comment를 수정한다. 수정 내용은 rating과 comment이다.
    * @param comment
    */
    void updateComment(Comment comment);
    
    /**
     * productId에 해당하는 Comment의 목록을 Comment id의 내림차순으로 반환한다.
     * @param productId
     * @return
     */
    List<Comment> selectByProduct(Integer productId);
}

File: app/app/src/main/java/com/ssafy/smartstoredb/data/service/CommentService.kt
package com.ssafy.smartstoredb.data.service

import com.ssafy.smartstore.util.RetrofitCallback
import com.ssafy.smartstore.util.RetrofitUtil
import com.ssafy.smartstoredb.model.dto.Comment
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class CommentService {
    // 코멘트 추가. response 받는 건 딱히없지 않나? 쓸모도 없을 것 같은데...
    fun addComment(comment: Comment, callback: RetrofitCallback<Boolean>)  {
        val menuInfoRequest: Call<Boolean> = RetrofitUtil.commentService.insert(comment)
        menuInfoRequest.enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if(response.code() == 200){
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }
        })
    }

    // 코멘트 업데이트.
    fun updateComment(comment: Comment, callback: RetrofitCallback<Boolean>)  {
        val menuInfoRequest: Call<Boolean> = RetrofitUtil.commentService.update(comment)
        menuInfoRequest.enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if(response.code() == 200){
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }
        })
    }

    // 코멘트 삭제
    fun removeComment(commentId: Int, callback: RetrofitCallback<Boolean>)  {
        val menuInfoRequest: Call<Boolean> = RetrofitUtil.commentService.delete(commentId)
        menuInfoRequest.enqueue(object : Callback<Boolean> {
            override fun onResponse(call: Call<Boolean>, response: Response<Boolean>) {
                val res = response.body()
                if(response.code() == 200){
                    if (res != null) {
                        callback.onSuccess(response.code(), res)
                    }
                } else {
                    callback.onFailure(response.code())
                }
            }

            override fun onFailure(call: Call<Boolean>, t: Throwable) {
                callback.onError(t)
            }
        })
    }
}